packages/console/src/common/Code.tsx

```diff
@@ -1,51 +1,39 @@
-import React, { Component, ReactElement, ReactNode } from 'react';
+import React, { useEffect, useRef, ReactNode, useCallback } from 'react';
 import * as monaco from 'monaco-editor';
+import { useTheme } from '@deephaven/components';
 
 interface CodeProps {
   children: ReactNode;
   language: string;
 }
 
-class Code extends Component<CodeProps, Record<string, never>> {
-  constructor(props: CodeProps) {
-    super(props);
-
-    this.container = null;
-  }
-
-  componentDidMount(): void {
-    this.colorize();
-  }
-
-  container: HTMLDivElement | null;
-
-  colorize(): void {
-    const { children } = this.props;
-    if (this.container && children != null) {
-      monaco.editor.colorizeElement(this.container, {
-        theme: 'dh-dark',
-      });
+function Code({ children, language }: CodeProps): JSX.Element {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const { activeThemes } = useTheme();
+
+  const colorize = useCallback(async () => {
+    if (containerRef.current && children != null) {
+      const result = await monaco.editor.colorize(
+        children.toString(),
+        language,
+        {}
+      );
+      containerRef.current.innerHTML = result;
     }
-  }
-
-  render(): ReactElement {
-    const { children, language } = this.props;
-    return (
-      <div>
-        <div
-          data-lang={language}
-          ref={container => {
-            this.container = container;
-          }}
-          // Add pointerEvents: 'none' has huge benefits on performance with Hit Test testing on large colorized elements.
-          // You can still select the text event with this set
-          style={{ pointerEvents: 'none' }}
-        >
-          {children}
-        </div>
-      </div>
-    );
-  }
+  }, [children, language]);
+
+  useEffect(() => {
+    if (activeThemes != null) colorize();
+  }, [activeThemes, colorize]);
```

Comment:
There's technically a race condition here - if you colorize a snippet, but before that returns the component changes and you colorize again, but the second colorize completes before the first one, it'll get set to the wrong value. Or it'll try to set it after an unmount. Should instead set an `isCanceled` flag and a `useEffect` straight up, e.g.

```typescript
  useEffect(() => {
    let isCanceled = false;
    async function colorize() {
      if (containerRef.current == null || children == null) {
        return;
      }
      const result = await monaco.editor.colorize(
        children.toString(),
        language,
        {}
      );
      if (isCanceled) {
        return;
      }
      containerRef.current.innerHTML = result;
    }
    colorize();
    return () => {
      isCanceled = true;
    };
  }, [activeThemes, children, language]);
```

Also - we should just use the `dangerouslySetInnerHTML` prop instead of using a container ref, e.g.
```

function Code({ children, language }: CodeProps): JSX.Element {
  const [colorizedHtml, setColorizedHtml] = useState<string | null>(null);
  const { activeThemes } = useTheme();

  useEffect(() => {
    let isCanceled = false;
    async function colorize() {
      if (children != null) {
        const result = await monaco.editor.colorize(
          children.toString(),
          language,
          {}
        );
        if (!isCanceled) {
          setColorizedHtml(result);
        }
      }
    }
    colorize();
    return () => {
      isCanceled = true;
    };
  }, [activeThemes, children, language]);

  return (
    <div
      // Add pointerEvents: 'none' has huge benefits on performance with Hit Test testing on large colorized elements.
      // You can still select the text event with this set
      style={{ pointerEvents: 'none' }}
      // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={
        colorizedHtml != null ? { __html: colorizedHtml } : undefined
      }
    />
  );
}
```

Reply:
totally forgot `dangerouslySetInnerHTML` exists.